test/protocols/binary_test.exs

```diff
@@ -1,80 +1,183 @@
 defmodule BinaryProtocolTest do
+  use ThriftTestCase, gen_erl: true
   use ExUnit.Case
 
-  @thrift_file_path "./test/fixtures/app/thrift/simple.thrift"
+  @moduletag :integration
 
-  import ParserUtils
+  def round_trip_struct(data, serializer_mf, deserializer_mf) do
+    {serializer_mod, serializer_fn} = serializer_mf
+    {deserializer_mod, deserializer_fn} = deserializer_mf
 
-  setup_all do
-    @thrift_file_path
-    |> parse_thrift
-    |> compile_module
+    serialized = :erlang.apply(serializer_mod, serializer_fn, [:struct, data])
+    |> IO.iodata_to_binary
 
-    :ok
+    :erlang.apply(deserializer_mod, deserializer_fn, [serialized])
   end
 
-  test "it should not encode empty fields" do
-    assert <<0>> == serialize_user(user(:elixir))
+  @thrift_file name: "enums.thrift", contents: """
+  enum Status {
+    ACTIVE,
+    INACTIVE,
+    BANNED = 6,
+    EVIL = 0x20,
+  }
+
+  struct StructWithEnum {
+    1: Status status
+  }
+  """
+  thrift_test "encoding enums" do
+    encoder = {StructWithEnum.BinaryProtocol, :serialize}
+    decoder = {Erlang.Enums, :deserialize_struct_with_enum}
```

Comment:
I wonder if namespacing will be an issue. The Elixir stuff is namespaced to this test module and aliased. I suppose you could do the same with the generated Erlang.* modules but the Erlang code generated by Thrift won't have any such concept. Should we just not worry about it for now?